articles/140_topic_and_service_name_mapping.md

```diff
@@ -265,68 +231,53 @@ The standard reserves the right to use up to 8 characters for the prefix in case
-| `/robot1/camera_left/image_raw` | `image_raw` | \[`rt/robot1/camera_left`\]   |
+| ROS Name                        | DDS Topic                         |
+|---------------------------------|-----------------------------------|
+| `/foo`                          | `rt/foo`                          |
+| `rostopic:///foo/bar`           | `rt/foo/bar`                      |
+| `/robot1/camera_left/image_raw` | `rt/robot1/camera_left/image_raw` |
 
-### Changing ROS Names During Runtime
-
-If the ROS name needs to be changed during runtime (after the publisher or subscriber has already been created and is in use), then some details about DDS partitions might be useful to consider.
-
-Since this proposal splits the ROS name into namespace and base name, we'll consider how that might affect DDS partitions and DDS topic names respectively.
-When changing the namespace of a ROS name, i.e. everything but the base name, then this could be done by changing just the existing partition string.
-As partitions are part of QoS settings, it technically doesn't require a full restart of the publisher and subscription when changing the partitions field, however we cannot change the DDS topic name on the fly.
-With this in mind, we may want to always require a full restart of the publisher or subscription instance to avoid inconsistency in how these changes behave from the ROS user's perspective.
 
 ### ROS Topic and Service Name Length Limit
 
-The length of the DDS topic must not exceed 256 characters.
-The actual length of a partition field may be limited to 256 characters, however this varies drastically depending on the vendor.
-RTI Connext does not allow a creation of a publisher/subscription with a partition length of 248 characters, whereas FastRTPS does not have any limitation in length.
-Please bare in mind, that the length of the partition gets further diminished due to the introduction of a ROS specific prefix.
-The actual length of a ROS Topic, including the namespace hierarchy and the base name of the topic, may thus be varying in length as well.
-Yet, the base name token must not exceed the length of 256 characters as this is getting mapped directly as the DDS topic.
+The length of the DDS topic must not exceed 256 characters. Therefore the length of a ROS Topic, including the namespace hierarchy, the base name of the topic and any ros specific prefixes must not exceed 256 characters since this is mapped directly as DDS topic.
+
+The length of service name has tighter limits than the length of the ROS Topics.
+In underlying RMW-DDS implementation, because the way in which services are implemented vary for different DDS vendors, and hence imposes the limit on actual service names that can be used.
+For example, in RTI connext, the service names are suffixed with the GUID value of the DDS participant, and a content filtered topic(max length 256 characters) is created which is mapped from the suffixed service name.
```

Comment:
That's odd. Where can I read more up on this about RTI Connext? I'd think this runtime specific GUID suffix would disrupt efforts to statically define topic permissions for DDS security.

Reply:
I believe that the suffix is not taken into account by Connext's the builtin plugins.
That is why the current sros2 generated permission files work for Connext even if they dont have the postfix in them. Not sure how they handle it internally though..